Jenkinsfile

```diff
@@ -1,182 +1,97 @@
 #!/usr/bin/env groovy
 
-REPOSITORY = "publishing-api"
-DEFAULT_SCHEMA_BRANCH = "deployed-to-production"
-DEFAULT_CONTENT_STORE_BRANCH = "deployed-to-production"
-DEFAULT_PUBLISHING_E2E_TESTS_BRANCH = "test-against"
-
 node {
   def govuk = load("/var/lib/jenkins/groovy_scripts/govuk_jenkinslib.groovy")
 
-  properties([
-    buildDiscarder(
-      logRotator(
-        numToKeepStr: "10"
-      )
-    ),
-    parameters([
-      booleanParam(
-        name: "IS_SCHEMA_TEST",
-        defaultValue: false,
-        description: "Identifies whether this build is being triggered to test a change to the content schemas"
-      ),
-      stringParam(
-        name: "SCHEMA_BRANCH",
-        defaultValue: DEFAULT_SCHEMA_BRANCH,
-        description: "The branch of govuk-content-schemas to test against"
-      ),
+  govuk.buildProject(
+    extraParameters: [
       stringParam(
         name: "CONTENT_STORE_BRANCH",
-        defaultValue: DEFAULT_CONTENT_STORE_BRANCH,
+        defaultValue: "deployed-to-production",
         description: "The branch of content-store to test pacts against"
       ),
       stringParam(
         name: "PUBLISHING_E2E_TESTS_BRANCH",
-        defaultValue: DEFAULT_PUBLISHING_E2E_TESTS_BRANCH,
+        defaultValue: "test-against",
         description: "The branch of publishing-e2e-tests to test against"
-      ),
-    ])
-  ])
-
-  try {
-    govuk.initializeParameters([
```

Comment:
I think we have an issue here, now that this isn't called, the parameters don't get set as environment variables the first time as per: https://github.com/alphagov/govuk-puppet/blob/5ad9ad4b597a76a3011ccc8ff9cdc03f81dd4384/modules/govuk_jenkins/files/var/lib/jenkins/groovy_scripts/govuk_jenkinslib.groovy#L378

So this means below where they are used `sh("git checkout ${env.CONTENT_STORE_BRANCH}")`, it's coming up as `null`, as can be seen in: https://ci.integration.publishing.service.gov.uk/job/publishing-api/job/use_rabbitmq_envvar/3/console

Reply:
Ah, interesting.  Do you think this this something we should add back into this Jenkinsfile, or add to `buildProject` 'cos it's a more widespread problem?

Reply:
I'm surprised this hasn't come up before, so I'm wondering how other teams are mitigating the problem. Adding it to `buildProject` feels like the right thing to do though.

Reply:
Yeah, I'll take a look - I think we can iterate over the existing params definitions and use their default values so we should be able to add it without extra config from the calling Jenkinsfile.

Reply:
@thomasleese I've got a PR in puppet here: https://github.com/alphagov/govuk-puppet/pull/6826 and I've used the jenkins replay feature to try it out on your failing build - https://ci.integration.publishing.service.gov.uk/job/publishing-api/job/use_rabbitmq_envvar/15/console is a passing one (it took me a while).